python/tvm/relay/frontend/pytorch.py

```diff
@@ -184,11 +182,9 @@ def _impl(inputs, input_types):
 def _concatenate(prelude):
     def tensor_array_concat(lst, axis):
         assert axis == 0, "Tensor array concat supported only for axis 0"
-        shape = _infer_type_with_prelude(prelude.hd(lst), prelude).shape
-        concat_shape = (Any(),) + tuple(shape[1:])
-
-        tensor_array = _map_tensor_array_constructor(lst, prelude, shape)
-        static_tensor_array_ops = StaticTensorArrayOps(prelude, "float32", concat_shape)
+        tensor_array, shape = _convert_to_tensor_array(lst, prelude)
+        concat_shape = (Any(),) + shape[1:]
+        static_tensor_array_ops = StaticTensorArrayOps(prelude, "float32", shape)
         static_tensor_array_ops.define_tensor_get_data(concat_shape)
 
         concat = prelude.get_var_static('tensor_array_concat', "float32", concat_shape)
```

Comment:
Also need to use shape here?

Reply:
fixed. So a rule of thumb is only `define_tensor_get_data` takes a new (concat-ed, stacked etc) shape?